pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/DeletedObjectDao.java

```diff
@@ -0,0 +1,201 @@
+package org.broadinstitute.dsm.db.dao;
+
+import static org.broadinstitute.ddp.db.TransactionWrapper.inTransaction;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.structure.DBElement;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.model.patch.DeleteType;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+@Slf4j
+public class DeletedObjectDao {
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_TISSUE = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)), ') from ddp_tissue where tissue_id = ? ;') into @sql "
+            + " from information_schema.columns where table_name = 'ddp_tissue'";
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_SM_ID = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)), ') from sm_id where sm_id_pk = ? ;') into @sql "
+            + " from information_schema.columns where table_name = 'sm_id'";
+
+    private static final String SQL_GET_JSON_OBJECT_FOR_ONC_HISTORY = "select concat('select json_object(', group_concat( "
+            + " concat(quote(column_name), ', ', column_name)),') from ddp_onc_history_detail where onc_history_detail_id = ? ;')"
+            + " into @sql from information_schema.columns where table_name = 'ddp_onc_history_detail'";
+    
```

Comment:
I made 3 different strings that look the same to avoid the risk of SQL injection, let me know if you find this unnecessary

Reply:
SQL injection issues usually come up when code is blindly concatenating strings or running raw strings as SQL.  I think this isn't doing that--the are parameterized values and user input only gets into statements via stmt.setFoo().  If you are concerned that this is using SQL to generate SQL, that's not an injection hazard per se because the only part of the generated SQL that comes from user values is already sanitized upstream.

Having said that, this is one reason why setting ids as ints/longs is important--if you can only set an id as an int value, there's no string value that could possibly cause problems.

Reply:
I see!! I am not really scared of creating the query and storing it in `@sql` but was more afraid of the other, more refactored approach (that I did not use) of having only one query and adding the table names and primary keys with string concatenations to it based on what we are deleting from. Unfortunately the `stmt.setString()` can not be used to add the table name and column name, just the values so it's either creating separating queries, doing some replacing in the query or just concatenations and I think this was the best choice.